core/src/test/scala/sttp/tapir/testdata/BuiltinTypenameCollisionCaseClass.scala

```diff
@@ -0,0 +1,46 @@
+package sttp.tapir.testdata
+
+sealed abstract class BuiltinTypenameCollisionCaseClass
+
+object BuiltinTypenameCollisionCaseClass {
+
+  case class Option(a: String) extends BuiltinTypenameCollisionCaseClass
+  case class Some(a: Int) extends BuiltinTypenameCollisionCaseClass
+  case class None(b: Boolean) extends BuiltinTypenameCollisionCaseClass
+  case class List(c: String) extends BuiltinTypenameCollisionCaseClass
+  case class Nil(b: String) extends BuiltinTypenameCollisionCaseClass
+  case class Map(d: Int) extends BuiltinTypenameCollisionCaseClass
+
+  /** can't make this work - it either can't derive schema, or when schema is explicit in companion like below, it fails with next
+    * alphabetical co-product part. Magnolia issue?
```

Comment:
ha interesting ... that might be sth in magnolia - what's the error?

maybe let's move this comment to a separate issue? and I'll merge this then. The PR looks great, thanks! :)

Reply:
Sure, thanks. I'll create an issue and remove this comment